lib/job_simulation.rb

```diff
@@ -0,0 +1,56 @@
+class JobSimulation
+  attr_reader :working, :waiting
+
+  def initialize(jobs_available, job_seekers)
+    @working = CSFun::Stack.new
+    @waiting = CSFun::Queue.new
+
+    job_seekers.times do |i|
+      @waiting.enqueue("Worker \##{i + 1}")
+    end
+
+    while @working.size != jobs_available
+      @working.push( @waiting.dequeue )
+    end
+  end
+
+  def cycle(num=rand(1..6))
```

Comment:
This changes the definition of the method `cycle` that I provided you (by indicating that there could now be an optional parameter). I see that this is actually a really cool way to allow us to provide the number to cycle - but that doesn't change that it alters the spec that was provided to you, which shouldn't be done :)